packages/pwa-kit-extension-sdk/src/configs/utils.ts

```diff
@@ -34,34 +34,53 @@ Handlebars.registerHelper('jsonStringify', (context) => JSON.stringify(context,
 // to issues with pathing because the current working directory for the loader isn't the same as the base project.
 // We can look to resolve this in the future as it would be nice to have a independant file for the template.
 const templateString = dedent`
+    import {getConfiguredExtensions} from '../../shared/utils/helpers'
     {{#if (isWeb @root.target)}}
     import loadable from '@loadable/component'
     {{/if}}
 
-    {{#each installed}}
+    {{#each configured}}
     {{#if (isNode @root.target)}}
-    import {{getInstanceName .}} from '{{.}}/setup-server'
+    import {{getInstanceName this.[0]}} from '{{this.[0]}}/setup-server'
     {{else}}
-    const {{getInstanceName .}}Loader = loadable.lib(() => import('{{.}}/setup-app'))
+    const {{getInstanceName this.[0]}}Loader = loadable.lib(() => import('{{this.[0]}}/setup-app'))
     {{/if}}
     {{/each}}
 
+    const imports = {
+        {{#each configured}}
+        {{#if (isNode @root.target)}}
+        '{{this.[0]}}': {{getInstanceName this.[0]}},
+        {{else}}
+        '{{this.[0]}}': {{getInstanceName this.[0]}}Loader,
+        {{/if}}
+        {{/each}}
+    }
+
     {{#if (isNode @root.target)}}
     const getApplicationExtensions = () => {
-        {{#if configured}}
-        return [{{#each configured}}new {{getInstanceName this.[0]}}({{{jsonStringify this.[1]}}}){{#if (isNotLast @index @root.configured.length)}}, {{/if}}{{/each}}]
-        {{else}}
-        return []
-        {{/if}}    
+        const configuredExtensions = getConfiguredExtensions()
+        if (!configuredExtensions) return []
+
+        return configuredExtensions.map((extension) => {
+            const packageName = extension[0]
+            const config = extension[1]
+            return new imports[packageName](config)
+        })
     }
     {{else}}
     const getApplicationExtensions = async () => {
-    	{{#if configured}}
-        const modules = await Promise.all([{{#each configured}}{{getInstanceName this.[0]}}Loader.load(){{#if (isNotLast @index @root.configured.length)}},{{/if}}{{/each}}])
-        return [{{#each configured}}new modules[{{@index}}].default({{{jsonStringify this.[1]}}}){{#if (isNotLast @index @root.configured.length)}}, {{/if}}{{/each}}]
-        {{else}}
-        return []
-        {{/if}}
+        const configuredExtensions = getConfiguredExtensions()
+        if (!configuredExtensions) return []
+
+        const modules = await Promise.all(configuredExtensions.map((extension) => {
+            const packageName = extension[0]
+            return imports[packageName].load()
+        }))
+        return configuredExtensions.map((extension, index) => {
+            const config = extension[1]
```

Comment:
If you want the style to match from the change in the above comment you can do something like
`const [,config] = extension`